vault/wizards/vault_send_wizard.py

```diff
@@ -0,0 +1,48 @@
+# © 2021 Florian Kantelberg - initOS GmbH
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import _, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class VaultSendWizard(models.TransientModel):
+    _name = "vault.send.wizard"
+    _description = _("Wizard to send another user a secret")
+
+    user_id = fields.Many2one(
+        "res.users",
+        "User",
+        required=True,
+        domain=[("active_key", "!=", False), ("inbox_enabled", "=", True)],
+    )
+    name = fields.Char(required=True)
+    public = fields.Char(related="user_id.active_key.public")
+    iv = fields.Char(required=True)
+    key_user = fields.Char(required=True)
+    key = fields.Char(required=True)
+    secret = fields.Char(required=True)
+    secret_file = fields.Char()
+    filename = fields.Char()
+
+    _sql_constraints = [
+        (
+            "value_check",
+            "CHECK(secret IS NOT NULL OR secret_file IS NOT NULL)",
+            _("No value found"),
+        ),
+    ]
+
+    def action_send(self):
+        self.ensure_one()
+        self.env["vault.inbox"].sudo().store_in_inbox(
+            self.name,
+            self.secret,
+            self.secret_file,
+            self.iv,
+            self.key_user,
```

Comment:
I think key_user is not required here, key instead. Old code ?

Reply:
`key` here is the aes key wrapped with the own public key. `key_user` is the key wrapped with the other users public key. Because you send it to the other user he should be able to decrypt it. See the js vault_controller.js for this wizard.

Reply:
Yes, I did.

![image](https://user-images.githubusercontent.com/19529533/111071665-83436800-84d7-11eb-8266-c578fadeade2.png)


In the js, key_user is not used I think. Moreover, the create inbox function does not get that argument:

https://github.com/OCA/server-auth/pull/254/files#diff-cb469858e7c43a9acaa02aab5a9ed768ad679a61f1d9a891e3e50acffd5d4699R66

Reply:
Does the other user has a public key? Have you logged in once with the other user?

Reply:
Ok, so the user should have filled in his key ?

So, the message should be a little bit more explicit no ?

Reply:
Yes I try to add a proper message. This works with the rsa key. You can only send to other users if they also have a key pair.

Reply:
Ok I added message and fixed the domain there because the `active_key` field wasn't searchable. Only usable users should show up now.